### PR TITLE
Update Travis config to test against Ruby 2.2.5 and 2.3.1, enable cache

### PR DESCRIPTION
**Pull request by krzysiek1507**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
-bundler_args: --without development production --quiet
+language: ruby
+
+cache:
+  bundler: true
+
+rvm:
+  - 2.3.1
+  - 2.2.5
+  - 2.1.10
```
Comment (damianlegawiec) [inline .travis.yml]: @krzysiek1507 we can skip 2.1.x :)

Comment (krzysiek1507) [inline .travis.yml]: @damianlegawiec spree [requires](https://github.com/spree/spree/blob/master/core/spree_core.gemspec#L11) ` >= 2.1.0` :)

Comment (damianlegawiec) [inline .travis.yml]: @krzysiek1507 currently yes, but master will switch to `rails 5` soon and we will drop ruby `2.1.x` support

Comment (krzysiek1507) [inline .travis.yml]: @damianlegawiec done

**Comment by krzysiek1507** [comment]
@damianlegawiec Travis is failing on mysql but it isn't related to this PR.

**Comment by krzysiek1507** [comment]
@damianlegawiec appraisal is removed. Everything should work, but we have take a look on MySQL tests.

**Comment by Taher-Ghaleb** [comment]
Hello,

Sorry, I know it's been a long time ago, but I am curious why caching hasn't been enabled before this (i.e., since December 17, 2014, when it became available to open source projects).

Thank you.

**Pull request merged**